src/main/java/io/curity/authenticator/netid/config/NetIdAccessConfig.java

```diff
@@ -45,7 +50,18 @@ public interface NetIdAccessConfig extends Configuration
     @Description("Sets the full path to the Net iD Access service.")
     String getPath();
 
-    HttpClient getHttpClient();
+    @Description("The Net iD Access service trust store.")
+    Optional<ServerTrustCryptoStore> getTrustStore();
+
+    @Description("The keystore that will be used in connections to Net iD Access service.")
+    Optional<ClientKeyCryptoStore> getKeyStore();
+
+    @DefaultBoolean(false)
+    @Description("Whether to use an HTTP connection to the Net iD Access service. Defaults to an HTTPS connection.")
+    Boolean getUseHttpConnection();
+
+    @Description("Deprecated. The HTTP client is no longer used by the plugin. Use truststore, keystore, and 'use HTTP connection' settings instead.")
+    Optional<HttpClient> getHttpClient();
```

Comment:
I didn't want to break existing configurations so I left the HTTP client as an optional config. I added a note about deprecation in the description annotation but for some reason, this description is not rendered in the admin UI.

Reply:
I think its better to remove it, we break config model when migrating from internal plugin in any case.